include/aspect/utilities.h

```diff
@@ -1060,6 +1066,41 @@ namespace aspect
      * entry in the list must match one of the allowed operations.
      */
     std::vector<Operator> create_model_operator_list(const std::vector<std::string> &operator_names);
+
+    /**
+    * A class that represents a point in a chosen coordinate system.
+    */
+    template <int dim>
+    class GeneralizedPoint
+    {
+      public:
+        GeneralizedPoint(Point<dim> position, const GeometryModel::Interface<dim> *geometry_model);
+
+        /**
```

Comment:
a second constructor with std::array and the enum might be useful (not possible if you also store the Point of course)

Reply:
Although I would like to do this, I do not think it is possible, because an enum is not enough to find or make a geometry model from it, which is needed for the conversion.